### PR TITLE
Switch to jest

### PR DESCRIPTION
**Pull request by fcollonval**

### CODE REVIEW

**Comment by jtpio** [comment]
Thanks @fcollonval.

Looks like the GitHub Actions are not triggered automatically in this repo (probably because it comes from a fork).

Just checked on a fork to trigger them: https://github.com/jtpio/widget-ts-cookiecutter/runs/2002599820

`hashFiles` seems to be interpreted as a cookiecutter variable.

**Comment by ianhi** [comment]
> hashFiles seems to be interpreted as a cookiecutter variable.

You can have the cookiecutter skip the github actions by adding 
```	
"_copy_without_render": [
		".github/workflows/*"
```
to cookiecutter.json

as in:
https://github.com/ianhi/matplotlib-extension-cookiecutter/blob/c68571852ccd7e18be8c9519adbc55bcc6ca4593/cookiecutter.json#L9-L10

**Comment by jtpio** [comment]
Nice, thanks @ianhi for the tip!

**Comment by ianhi** [comment]
@jtpio it looks like the test now runs seemingly forever - currently at 1.5 hours :( https://github.com/jtpio/widget-ts-cookiecutter/runs/2005714585?check_suite_focus=true

github actions isn't showing me the text to see where it got stuck but running locally it seems to hang up on `npm run test:ci`

image

any idea why that is?

**Comment by jtpio** [comment]
> any idea why that is?

Maybe a karma timeout?

Also I was wondering whether we will want to stick to karma in the long run. If not we could then extract the GitHub Actions part of this PR to a separate PR to get it merged first.

Then we might want to switch to something else like `jest` which is for example used in JupyterLab.

**Comment by ianhi** [comment]
It looks karma start just starts the server but maybe doesn't actually run the tests https://stackoverflow.com/questions/19913934/difference-between-karma-start-and-run-karma maybe @fcollonval knows?

> Then we might want to switch to something else like jest which is for example used in JupyterLab.

It would definitely be nice to have similariteis to jlab and whatever the jlab-extension cookiecutter does.

**Comment by fcollonval** [comment]
Hey thank you both to push this one forward.

I would definitely like to have karma replace by jest. To be fair, I try a couple of time to make it work without success :cry: And I'm tired to have to fight with JS test framework configurations. So for private repos, I switched to jest.

So we can indeed put the switch to GitHub workflow in another PR - but karma is not fixed on travis neither :stuck_out_tongue_winking_eye: 

> Aside of the ipywidgets, another widget project that is working with karma is https://github.com/vidartf/ipydatawidgets/tree/master/packages/jupyter-datawidgets

**Comment by fcollonval** [comment]
Fixes #92

**Comment by fcollonval** [comment]
@jtpio @ianhi I switched to jest and restore all tests including the doc ones.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/home/fcollonval/miniconda3/envs/widgets/bin/python"
```
Comment (jtpio) [inline .vscode/settings.json:2]: We could probably add `.vscode` (and others like `.ide`) to the `.gitignore`.

Comment (fcollonval) [reply .vscode/settings.json]: Oups, sorry for than one

Comment (jtpio) [reply .vscode/settings.json]: no problem, thanks!

**Review by jtpio — APPROVED** [review]
Looks good, thanks for fixing this!

**Pull request merged**